pkg/manager/metricscollector/meticscollector.go

```diff
@@ -34,8 +35,16 @@ func NewMetricsCollector() (*MetricsCollector, error) {
 
 }
 
-func (d *MetricsCollector) CollectWorkerLog(wID string, objectiveValueName string, metrics []string, namespace string) (*api.MetricsLogSet, error) {
-	pl, _ := d.clientset.CoreV1().Pods(namespace).List(metav1.ListOptions{LabelSelector: "job-name=" + wID, IncludeUninitialized: true})
+func (d *MetricsCollector) CollectWorkerLog(wID string, wkind string, objectiveValueName string, metrics []string, namespace string) (*api.MetricsLogSet, error) {
+	var labelName string
+	if wkind == studyjob.TFJobWorker {
+		labelName = "tf_job_name"
+	} else if wkind == studyjob.PyTorchJobWorker {
+		labelName = "pytorch_job_name"
+	} else {
+		labelName = "job-name"
+	}
+	pl, _ := d.clientset.CoreV1().Pods(namespace).List(metav1.ListOptions{LabelSelector: labelName + "=" + wID, IncludeUninitialized: true})
```

Comment:
Whitch pod will be watched in Pytorch and TFJob Job by `pytorch_job_name = wID` and `tf_job_name  = wID` ?

Reply:
Pods spawn by the pytorch job and tf job will have this label key whose value is set to job name https://github.com/kubeflow/tf-operator/blob/master/pkg/common/jobcontroller/jobcontroller.go#L190

Reply:
Yes, my question is pytorch job and tf job will create several pods, and metrics collector will get logs from one pod. 
Does metrics collector get logs from which one of the pods created by a pytorch job?

Reply:
Currently, it uses only 1 worker. If there are more workers,  Master should take responsibility to emit logs.